plugins/package-managers/composer/src/main/kotlin/Composer.kt

```diff
@@ -248,6 +248,9 @@ class Composer(
         requireLockfile(workingDir) { hasLockfile }
         if (hasLockfile) return lockfile
 
+        // Ensure that the build is not configured to disallow the creation of lockfiles.
```

Comment:
commit-msg: Do I understand it correctly, that prior to this change, the composer analysis does not work
if `lock=false` is specified in `composer.json` and no lockfile exists`

Reply:
Correct. It would fail with

> Composer failed to resolve dependencies for path 'composer.json': FileNotFoundException: /tmp/ort-AnalyzerDownloader-analyzer-worker3019164139991636861/composer.lock (No such file or directory)

even with `allowDynamicVersions = true`.

Reply:
> Correct. Analysis would fail with

I'm totally fine with the change. But I believe this can be made easier to grasp in the commit message. How about starting the message body with describing this bug - and then explain how why this fixed it?

Reply:
I've tried my best to make it yet clearer.